Framework/PythonInterface/test/python/mantid/api/WorkspaceGroupTest.py

```diff
@@ -2,14 +2,74 @@
 
 import unittest
 from testhelpers import run_algorithm
-from mantid.api import mtd, WorkspaceGroup, MatrixWorkspace
+from mantid.api import mtd, WorkspaceGroup, MatrixWorkspace, AnalysisDataService, WorkspaceFactory
+
 
 class WorkspaceGroupTest(unittest.TestCase):
 
+    def create_matrix_workspace_in_ADS(self, name):
+        run_algorithm('CreateWorkspace', OutputWorkspace=name, DataX=[1., 2., 3.], DataY=[2., 3.], DataE=[2., 3.],
+                      UnitX='TOF')
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_WorkspaceGroup_can_be_instantiated(self):
+        ws_group = WorkspaceGroup()
+        self.assertIsNotNone(ws_group)
+        self.assertIsInstance(ws_group, WorkspaceGroup)
+
+    def test_that_instantiated_WorkspaceGroup_is_not_added_to_the_ADS(self):
+        AnalysisDataService.clear()
+
+        ws_group = WorkspaceGroup()
+
+        self.assertEqual(len(AnalysisDataService.getObjectNames()), 0)
+
+    def test_that_instantiated_WorkspaceGroup_can_be_added_to_the_ADS(self):
+        AnalysisDataService.clear()
+
+        ws_group = WorkspaceGroup()
+        mtd.add("group1", ws_group)
+
+        self.assertEqual(AnalysisDataService.getObjectNames(), ["group1"])
+        self.assertIsInstance(mtd["group1"], WorkspaceGroup)
+
+    def test_that_can_add_workspaces_to_WorkspaceGroup_when_in_ADS(self):
+        AnalysisDataService.clear()
+
+        self.create_matrix_workspace_in_ADS("ws1")
+        self.create_matrix_workspace_in_ADS("ws2")
+
+        ws_group = WorkspaceGroup()
+        mtd.add("group1", ws_group)
+
+        ws_group.add("ws1")
+        ws_group.add("ws2")
+
+        self.assertTrue("ws1" in mtd["group1"])
+        self.assertTrue("ws2" in mtd["group1"])
+
+    def test_that_can_add_workspaces_to_WorkspaceGroup_when_not_in_ADS(self):
+        AnalysisDataService.clear()
+
+        ws1 = WorkspaceFactory.create("Workspace2D", 2, 2, 2)
+        ws2 = WorkspaceFactory.create("Workspace2D", 2, 2, 2)
+
+        ws_group = WorkspaceGroup()
+
+        ws_group.addWorkspace(ws1)
+        ws_group.addWorkspace(ws2)
+
+        self.assertEqual(ws_group.size(), 2)
+
     def test_group_interface(self):
-        run_algorithm('CreateWorkspace', OutputWorkspace='First',DataX=[1.,2.,3.], DataY=[2.,3.], DataE=[2.,3.],UnitX='TOF')
-        run_algorithm('CreateWorkspace', OutputWorkspace='Second',DataX=[1.,2.,3.], DataY=[2.,3.], DataE=[2.,3.],UnitX='TOF')
-        run_algorithm('GroupWorkspaces',InputWorkspaces='First,Second',OutputWorkspace='grouped')
+        run_algorithm('CreateWorkspace', OutputWorkspace='First', DataX=[1., 2., 3.], DataY=[2., 3.], DataE=[2., 3.],
```

Comment:
Can we convert these to use your new `create_matrix_workspace_in_ADS` method for consistency across the suite?

Reply:
I will go through the other tests and neaten them up a bit.